assets/js/blocks-registry/payment-methods/test/registry.ts

```diff
@@ -0,0 +1,97 @@
+/**
+ * External dependencies
+ */
+import { registerPaymentMethodExtensionCallbacks } from '@woocommerce/blocks-registry';
+
+/**
+ * Internal dependencies
+ */
+import { canMakePaymentExtensionsCallbacks } from '../extensions-config';
+
+describe( 'registerPaymentMethodExtensionCallbacks', () => {
+	it( 'Logs an error to console if namespace is already registered', () => {
+		registerPaymentMethodExtensionCallbacks(
+			'woocommerce-marketplace-extension',
+			{
+				cod: () => false,
+			}
+		);
+
+		// eslint-disable-next-line no-console
+		expect( console ).not.toHaveErrored();
+		registerPaymentMethodExtensionCallbacks(
+			'woocommerce-marketplace-extension',
+			{
+				cod: () => false,
+			}
+		);
+		expect( console ).toHaveErrored();
+
+		// eslint-disable-next-line no-console
+		expect( console.error ).toHaveBeenCalledTimes( 1 );
+	} );
+
+	it( 'Does not overwrite a namespace if a second extensions tries to register with the same name', () => {
+		const firstCodCallback = jest.fn().mockReturnValue( false );
+		registerPaymentMethodExtensionCallbacks(
+			'overwrite-marketplace-extension',
+			{
+				cod: firstCodCallback,
+			}
+		);
+
+		// eslint-disable-next-line no-console
+		expect( console ).not.toHaveErrored();
+		registerPaymentMethodExtensionCallbacks(
+			'overwrite-marketplace-extension',
+			{
+				cod: () => false,
+			}
+		);
+
+		expect(
+			canMakePaymentExtensionsCallbacks[
+				'overwrite-marketplace-extension'
+			].cod
+		).toEqual( firstCodCallback );
+	} );
```

Comment:
@ralucaStan thanks for the review - this test checks that the namespace is not overwritten. I need to leave the `expect( console ).not.toHaveErrored();` line in because if not then the test case fails because of "unexpected" console errors 🤓